drv/opentitan-aontimer/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#![no_std]
+
+use core::ptr::slice_from_raw_parts_mut;
+
+const WDOG_REGWEN_IDX: usize = 4;
+const WDOG_CTRL_IDX: usize = 5;
+const WDOG_BARK_THOLD_IDX: usize = 6;
+const WDOG_BITE_THOLD_IDX: usize = 7;
+const WDOG_COUNT_IDX: usize = 8;
+const INTR_STATE_IDX: usize = 9;
+
+// The OpenTitan Always-On Timer is a counter that increments. If the counter
+// reaches a configurable threshold, it will emit an interrupt signal, called the
+// "bark". If the counter reaches a second configurable threshold, it will reset
+// the system, called the "bite".
+// Enabling the AON Timer before setting a bite threshold causes the system to
+// immediately reset.
+pub struct AonTimer {
+    base_addr: *mut [u32],
+    clock_freq_hz: u32,
+    pub bark_cb: Option<fn()>,
+}
+
+impl AonTimer {
+    // Initialize a new watchdog driver device with the specified base address,
```

Comment:
Reading the [AON timer spec](https://docs.opentitan.org/hw/ip/aon_timer/doc/), the AON timer device supports 2 timers - watchdog timer and wakeup timer. It looks like this patch adds support for watchdog timer, but not the wakeup timer? However, the following patch for demo actually makes use of both the watchdog timer as well as the wakeup timer. Seems like there is a disconnect?

Reply:
I think I figured out my source of confusion here. As mentioned above there are two timers within the AON timer device. The demo app is configuring and using both timers. But, the wakeup timer is handled by the kernel in hubris, whereas the watchdog timer is handled by the AON timer driver.